src/record/recordUpdater.test.ts

```diff
@@ -11,10 +11,54 @@ import { User } from '../auth'
 import { Records } from './records'
 import { Record } from './record'
 import { Validations } from '../validation/validations'
-import { Surveys } from '../survey'
+import { Survey } from '../survey'
 
 let user: User
 
+const updateAttributeAndExpectValidation = async (params: {
+  survey: Survey
+  record: Record
+  nodePath: string
+  value: any
+  validationNodePath?: string
+  expectedFieldValidation: boolean
+  expectedValidationFieldsSize: number
+}): Promise<Record> => {
+  const {
+    survey,
+    record,
+    nodePath,
+    value,
+    validationNodePath = params.nodePath,
+    expectedFieldValidation,
+    expectedValidationFieldsSize,
+  } = params
+  const nodeToUpdate = TestUtils.getNodeByPath({ survey, record, path: nodePath })
+
+  const nodeUpdated = { ...nodeToUpdate, value }
+  const recordUpdated = Records.addNode(nodeUpdated)(record)
```

Comment:
why did you add the node to the record before using the updateNode?

Reply:
fixed now